utils/upload_ansible_roles_to_galaxy.py

```diff
@@ -135,12 +136,57 @@ def gather_data(self):
                     "pre_tasks are not supported for ansible roles and "
                     "will be skipped!.\n")
 
-        self.tasks_local_content = yaml.dump(
-            self.tasks_data, width=120, indent=4, default_flow_style=False)
-
         description = self.get_description_from_filedata(filedata)
         self.description = description
 
+    def add_variables_to_tasks(self):
+        for task in self.tasks_data:
+            if "when" not in task:
+                task["when"] = []
+            elif isinstance(task["when"], str):
+                task["when"] = [task["when"]]
+
+            self.add_variables_to_task(task)
+
+            if not task["when"]:
+                del task["when"]
+
+    def tag_is_valid_variable(self, tag):
+        if "-" in tag:
+            return False
+        if tag == "always":
+            return False
+        return True
+
+    def add_variables_to_task(self, task):
+        if "tags" not in task:
+            return
+        variables_to_add = {tag for tag in task["tags"] if self.tag_is_valid_variable(tag)}
+        task["when"] += ["{{{{ {varname} | bool }}}}".format(varname=v) for v in variables_to_add]
+        self.added_variables.update(variables_to_add)
+
+    def add_task_variables_to_default_variables_if_needed(self):
+        default_vars_to_add = sorted(self.added_variables)
+        lines = [
+            "---", "# defaults file for {role_name}\n".format(role_name=self.remote_repo.name),
+        ]
+        lines += ["{var_name}: true".format(var_name=var_name) for var_name in default_vars_to_add]
```

Comment:
It creates a variable in single quotes`'`. Shouldn't it be in double quotes `"`?

Reply:
I have found out that it is the lower level of the code that applies single quotes, as it detects that an item is a string, so it is quoted. In this case, the difference between single and double quotes is not important, as it has something to do with escaping non-ASCII chars.
Therefore, I propose to ignore this matter.

Reply:
@matejak Aha OK. But a different way of question: Does Ansible expand the Jinja 2 templates if the templates are inside a single-quoted string?

Reply:
I think so. In the [Ansible documentation](https://docs.ansible.com/ansible/2.5/user_guide/playbooks_filters.html#combination-filters), there is pointed out that jinja2 is expanded behind double quotes. And there is no indication anywhere that single quotes are treated differently. And it also is the expected behavior - jinja2 looks for curly braces, not being disturbed by any other chars, and YAML vs quotes has been touched in my previous comment.